File: 1046.py
num1,num2 = input().split()
num1,num2 = int(num1),int(num2)
duracao = 0
if num1 > num2:
    duracao= 24 + num2 - num1
elif num2 > num1:
    duracao= num2 - num1
else:
    duracao=24
print(f"O JOGO DUROU {duracao} HORA(S)")

File: cpp/1046.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){

    int hi, hf; cin >> hi >> hf;
    int cont =0;

    if (hi == hf){
        cout << "O JOGO DUROU " << 24 << " HORA(S)" << "\n";
    }

    else{
        while (hi != hf){
            hi++;
            cont++;
            if(hi > 23){
                hi=0;
            }
        }
    
        cout << "O JOGO DUROU " << cont << " HORA(S)" << "\n";
    }

}